### PR TITLE
Split Blockly#blockly_options into blockly_app_options and blockly_level_options

### PR DESCRIPTION
**Pull request by balderdash**
Working toward generating `blockly_level_options` multiple times.

### CODE REVIEW

**Comment by Hamms** [comment]
I love seeing a refactor like this that ultimately adds so few lines!

Didja talk to Will about the current app/level options split?

File: dashboard/app/models/blockly.rb [hunk 1]
```diff
@@ -202,10 +202,27 @@ def blocks_to_embed(block_xml)
...
+      app_options[:levelGameName] = game.name if game
+      app_options[:skin_id] = skin_id if skin_id
+
+      # Set some values that Blockly expects on the root of its options string
+      app_options.merge!({
+                             baseUrl: Blockly.base_url,
+                             app: game.try(:app),
+                             droplet: game.try(:uses_droplet?),
+                             pretty: Rails.configuration.pretty_apps ? '' : '.min',
+                         })
```
Comment (Hamms) [inline dashboard/app/models/blockly.rb]: I believe our styleguide prefers these to be all left-aligned:

```
app_options.merge!({
  baseUrl: Blockly.base_url,
  app: game.try(:app),
  droplet: game.try(:uses_droplet?),
  pretty: Rails.configuration.pretty_apps ? '' : '.min',
})
```

File: dashboard/app/models/blockly.rb [hunk 2]
```diff
@@ -270,23 +287,9 @@ def blockly_options
...
-      app_options[:levelGameName] = level.game.name if level.game
-      app_options[:skinId] = level.skin if level.is_a?(Blockly)
-
       # Set some values that Blockly expects on the root of its options string
-      non_nil_level_prop = level_prop.reject!{|_, value| value.nil?}
-      app_options.merge!({
-                             baseUrl: Blockly.base_url,
-                             app: level.game.try(:app),
-                             levelId: level.level_num,
-                             level: non_nil_level_prop,
```
Comment (Hamms) [inline dashboard/app/models/blockly.rb:284]: Why were these removed?

Edit: by which I mostly mean, how are we confident that nothing was using these?

**Comment by Hamms** [comment]
This look quite good, but I'd love to chat briefly about the motivation behind this change just to make sure I actually understand the context.

File: dashboard/app/models/blockly.rb (hunk 2 above)
Comment (balderdash) [inline dashboard/app/models/blockly.rb:284]: `:level` is being set in levels_helper.rb.

I think I misread https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/appMain.js#L29 as making `levelId` unnecessary, but that doesn't seem to be the case. I just added `:levelId` alongside `:level` in levels_helper.rb, but I'll need to move it into `blockly_level_options` and refactor the JS a a bit (in another PR).

**Comment by balderdash** [comment]
Talked to Will about the `view_options`/`level_view_options` split a bit. It seems the original intent for the split has degraded a bit as people added config options, which sounds about right to me. I'm just going to continue cleaning that up.

As a sanity check, I compared appOptions before and after this change on a random level (/course4/stage/16/puzzle/4). Aside from timestamps and a couple props being reordered, they're identical.

**Comment by Hamms** [comment]
LGTM!
